persistence-jdbc/core/src/main/scala/com/lightbend/lagom/internal/persistence/LagomDBApiProvider.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+package com.lightbend.lagom.internal.persistence
+
+import com.typesafe.config.Config
+import javax.inject.{ Inject, Singleton }
+import play.api.db.{ ConnectionPool, DBApi, DBApiProvider, DefaultDBApi }
+import play.api.inject.{ ApplicationLifecycle, Injector, NewInstanceInjector }
+import play.api.{ Configuration, Environment, Logger }
+
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+// This overrides the DBApiProvider from Play with Lagom-specific behavior
+// TODO: file an issue in Play to make this configurable so we can remove the override
+@Singleton
+private[lagom] class LagomDBApiProvider @Inject() (
+  environment:           Environment,
+  configuration:         Configuration,
+  defaultConnectionPool: ConnectionPool,
+  lifecycle:             ApplicationLifecycle,
+  injector:              Injector             = NewInstanceInjector
+) extends DBApiProvider(
+  environment,
+  configuration,
+  defaultConnectionPool,
+  lifecycle,
+  injector
+) {
+
+  private val logger = Logger(classOf[LagomDBApiProvider])
+
+  override lazy val get: DBApi = {
+    val config = configuration.underlying
+    val dbKey = config.getString("play.db.config")
+    val pool = ConnectionPool.fromConfig(config.getString("play.db.pool"), injector, environment, defaultConnectionPool)
+    val configs = if (config.hasPath(dbKey)) {
+      Configuration(config).getPrototypedMap(dbKey, "play.db.prototype").mapValues(_.underlying)
+    } else Map.empty[String, Config]
+    val db = new DefaultDBApi(configs, pool, environment, injector)
+    lifecycle.addStopHook { () =>
+      Future.successful {
+        try db.shutdown()
+        catch {
+          case NonFatal(ex) => logger.error("error on db shutdown", ex)
```

Comment:
In a real-world running application, we may also have that same situation. We stop the app, but the db is unreachable for some reason. In such a case there is not much one can do.

The reason why I'm logging it instead of just let the exception blow up everything is that I want to prevent the tests to fail. Without that, `application.stop` will return a failed `Future`.

WDYT?